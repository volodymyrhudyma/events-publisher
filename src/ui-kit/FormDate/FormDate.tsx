import React, { ChangeEvent } from 'react';
import { IFormControlProps } from 'src/types';
import FormControl from 'src/ui-kit/FormControl';
import Input from 'src/ui-kit/Input';
import Label from 'src/ui-kit/Label';
import Radio from 'src/ui-kit/Radio';
import { getCurrentDate } from 'src/utils/date';
import { TimeTypeWrapper, TipWrapper, Wrapper } from './styled/FormDate.styled';

interface IFormDateProps extends IFormControlProps {
  placeholder: string;
  tip: string;
  dateValue: string;
  timeValue: string;
  timeTypeValue: string;
  onDateChange: (e: ChangeEvent<HTMLInputElement>) => void;
  onTimeChange: (e: ChangeEvent<HTMLInputElement>) => void;
  onTimeTypeChange: (type: string) => void;
}

const FormDate = ({
  placeholder,
  tip,
  dateValue,
  timeValue,
  timeTypeValue,
  onDateChange,
  onTimeChange,
  onTimeTypeChange,
  ...rest
}: IFormDateProps) => (
  <FormControl {...rest}>
    <Wrapper>
      <Input
        type="date"
        placeholder={placeholder}
        width={180}
        value={dateValue}
        onChange={onDateChange}
        min={getCurrentDate()}
        error={rest.error}
      />
      <TipWrapper>
        <Label color="secondary" uppercase={false}>
          {tip}
        </Label>
      </TipWrapper>
      <Input
        type="time"
        placeholder={placeholder}
        width={100}
        value={timeValue}
        onChange={onTimeChange}
        min="00:00"
        max="12:00"
        error={rest.error}
      />
      <TimeTypeWrapper>
        <Radio
          value="am"
          label="AM"
          checked={timeTypeValue === 'am'}
          onChange={onTimeTypeChange}
        />
        <Radio
          value="pm"
          label="PM"
          checked={timeTypeValue === 'pm'}
          onChange={onTimeTypeChange}
        />
      </TimeTypeWrapper>
    </Wrapper>
  </FormControl>
);

FormDate.defaultProps = {
  required: false,
};

export default FormDate;
