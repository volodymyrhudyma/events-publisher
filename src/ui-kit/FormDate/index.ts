import Component from './FormDate';

export default Component;
