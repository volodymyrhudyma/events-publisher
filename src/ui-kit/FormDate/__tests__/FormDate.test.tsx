import React from 'react';
import { shallow } from 'enzyme';
import FormDate from '../FormDate';

describe('FormDate component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <FormDate
        label="Label"
        placeholder="Placeholder"
        tip="Tip"
        dateValue=""
        timeValue=""
        timeTypeValue=""
        onDateChange={jest.fn()}
        onTimeChange={jest.fn()}
        onTimeTypeChange={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
