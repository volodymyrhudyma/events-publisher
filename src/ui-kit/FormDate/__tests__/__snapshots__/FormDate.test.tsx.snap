// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormDate component should render correctly 1`] = `
<FormControl
  label="Label"
  required={false}
>
  <styled.div>
    <Input
      min="2019-05-28"
      onChange={[MockFunction]}
      placeholder="Placeholder"
      type="date"
      value=""
      width={180}
    />
    <styled.div>
      <Label
        color="secondary"
        uppercase={false}
      >
        Tip
      </Label>
    </styled.div>
    <Input
      max="12:00"
      min="00:00"
      onChange={[MockFunction]}
      placeholder="Placeholder"
      type="time"
      value=""
      width={100}
    />
    <styled.div>
      <Radio
        checked={false}
        label="AM"
        onChange={[MockFunction]}
        value="am"
      />
      <Radio
        checked={false}
        label="PM"
        onChange={[MockFunction]}
        value="pm"
      />
    </styled.div>
  </styled.div>
</FormControl>
`;
